src/main/java/lotto/domain/LottoCalculator.java

```diff
@@ -0,0 +1,22 @@
+package lotto.domain;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class LottoCalculator {
+
+    public LottoResult calculate(int purchaseAmount, List<Lotto> lottoList, List<Integer> winningNumbers) {
+        long totalPrize = 0;
+        Map<Integer, Integer> matchCounts = new HashMap<>();
```

Comment:
```suggestion
        Map<LottoPrize, Integer> matchCounts = new HashMap<>();
```

이렇게 Map 을 구성하는것도 좋은 방법일것 같아요